openmmtools/alchemy.py

```diff
@@ -473,35 +971,42 @@ def _buildAlchemicalTorsionList(self, alchemical_atomset):
         """
 
         # Tabulate all bonds
-        bonds = self._tabulateBonds(self.reference_system)
-        def is_bonded(i,j):
+        bonds = cls._tabulate_bonds(system)
+
+        def is_bonded(i, j):
             if j in bonds[i]:
                 return True
             return False
-        def is_proper_torsion(i,j,k,l):
-            if is_bonded(i,j) and is_bonded(j,k) and is_bonded(k,l):
+
+        def is_proper_torsion(i, j, k, l):
```

Comment:
@jchodera may be able to answer this question (it was his original implementation I believe).